pkg/util/response/response.go

```diff
@@ -0,0 +1,51 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package response
+
+import (
+	"github.com/go-openapi/runtime"
+	httptransport "github.com/go-openapi/runtime/client"
+
+	"github.com/elastic/cloud-sdk-go/pkg/client"
+)
+
+// SetRawJSON prepares the API transport to send raw JSON
+//
+// While setting the global JSONMime producer to runtime.TextProducer() looks
+// scary, runtime.TextProducer() falls back on swag.WriteJSON(data) if the
+// object we're marshalling is a struct or slice. So it actually works in most
+// cases.
+//
+// This is only necessary because the Cloud Swagger Spec lists some JSON blob
+// data as strings. When those are changed to JSON Objects, this can go away.
+func SetRawJSON(c *client.Rest) {
```

Comment:
Is is too much to ask for a couple of UTs here? 😊

Reply:
Added some small tests :D 

In order to be able to test every line I would have to create some  functions to mock the rest client in the `mock` package. This work, imo, falls out of the scope of this PR which is part of the cloud-cli layout structure change issue. To be honest, only these two functions would need said mocks and these are already tested [here](https://github.com/elastic/cloud-cli/blob/master/app/deployment/elasticsearch/metadata/raw_test.go#L111).

Unless I'm missing some super simple way to further test this?

Reply:
If you pass a `&client.Rest{Transport: &httptransport.Runtime{}}`, you can then assert that the values of it have been modified, that's the only test needed here

Reply:
so erm, I get a panic when I try to pass `&client.Rest{Transport: &httptransport.Runtime{}}`

```
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 9 [running]:
testing.tRunner.func1(0xc000490300)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x17f9080, 0x19fc9d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/elastic/cloud-sdk-go/pkg/util/response.SetRawJSON(0xc0000ee6c0)
	/Users/karen/go/src/github.com/elastic/cloud-sdk-go/pkg/util/response/response.go:39 +0xa0
...
```

Reply:
the `Producers` field (map) needs to be populated, it's complaining that an assignment is being made to a nil map, which it is (`panic: assignment to entry in nil map`).